src/proto/message/unparsed.rs

```diff
@@ -16,6 +16,11 @@ impl Unparsed {
         let mut v = &self.0[..];
         T::decode(&mut v)
     }
+
+    /// Expose the inner content in raw format.
+    pub fn as_slice(&self) -> &[u8] {
```

Comment:
I would be happy to add an `hazmat` feature if needed, or rename it like `as_raw` or something to highlight this is dangerous, but then I would argue `From<Vec<u8>>` should be removed too and replaced by:
```
impl Unparsed {
   pub fn new<T>(v: T) where T: ssh_encoding::Encoding -> Result<Self> {
   }
}
```

to highlight that.

Reply:
Okay, I like the "as raw" naming. Maybe to keep the symmetry we'd need "from raw" which takes a vec of unparsed data and doesn't interpret it in any way?

I like this new constructor. Do I read the signature well that it is intended for ssh encod-able data not for raw stuff?

I mean... we need raw support for crazy folks that want to use serde there (just when we got rid of it 😉).

One potential addition, not sure if good, is implementing `AsRef<[u8]>`. This would allow treating Unparsed as byte buffers (the implementation is the same as what you have here, just... more idiomatic (?)).

Reply:
`AsRef<[u8]>` makes accessing the content not-so-explicit imho, which renders the use of a type system a bit moot.